deploy-agent/deployd/common/executor.py

```diff
@@ -182,10 +182,21 @@ def _get_sleep_interval(self, start, interval):
     @staticmethod
     def _kill_process(process):
         try:
+            log.info("Kill currently running process")
             os.killpg(process.pid, signal.SIGKILL)
         except Exception as e:
             log.debug('Failed to kill process: {}'.format(e))
 
+    @staticmethod
+    def _graceful_shutdown(process):
+        try:
+            log.info("Gracefully shutdown currently running process")
+            os.killpg(process.pid, signal.SIGTERM)
+            process.wait(timeout=30)
```

Comment:
Should we make this configurable? Perhaps a new config like the other ones defined in __init__?

Reply:
I was thinking about it but is it really needed? Cleanup should happen in reasonable time?

Reply:
I think it should be configurable so that it does not need a deployd rollout if we need to change it.